util/chunk/alloc.go

```diff
@@ -139,6 +139,24 @@ func (a *allocator) Reset() {
 	}
 }
 
+// checkColumnType check whether the conditions for entering the corresponding queue are met
+// column Reset may change type
+func checkColumnType(id int, col *Column) bool {
+	if col.avoidReusing {
+		return false
+	}
+
+	if id == varElemLen {
+		//Take up too much memory,
+		if cap(col.data) > MaxCachedLen {
+			return false
+		}
+		return col.elemBuf == nil
+	}
+
+	return id == cap(col.elemBuf)
```

Comment:
s/id/sizeClass?

When this may differ? caused by exchange column?

Reply:
(c *Column) Reset can change column type